Dockerfile

```diff
@@ -1,79 +1,79 @@
-ARG DEBIAN_VERSION=buster-slim
-
 ############################
 # STEP 1 build executable binary
 ############################
 
-FROM debian:$DEBIAN_VERSION as builder
+FROM alpine:latest as builder
 
-RUN apt-get update && apt-get install -y \
-    build-essential \
-    git-core \
+RUN apk add --no-cache \
+    gcc \
+    g++ \
+    libc-dev \
+    linux-headers \
+    git \
+    make \
     cmake \
-    pkg-config \
-    libcurl4-openssl-dev \
-    libgnutls28-dev \
-    libsasl2-dev \
-    uuid-dev \
+    curl-dev \
+    gnutls-dev \
+    cyrus-sasl-dev \
+    # for libuuid
+    util-linux-dev \
     libtool \
-    libssl-dev \
-    libgcrypt20-dev \
+    libgcrypt-dev \
     libmicrohttpd-dev \
-    libltdl-dev \
-    libjson-c-dev \
-    libleptonica-dev \
-    libmosquitto-dev \
+    json-c-dev \
+    mosquitto-dev \
     libunistring-dev \
-    dh-autoreconf \
-    && rm -rf /var/lib/apt/lists/*
+    automake \
+    autoconf
 
 WORKDIR /vzlogger
 
 RUN git clone https://github.com/volkszaehler/libsml.git --depth 1 \
- && make install -C libsml/sml
+    && make install -C libsml/sml
 
 RUN git clone https://github.com/rscada/libmbus.git --depth 1 \
- && cd libmbus \
- && ./build.sh \
- && make install
+    && cd libmbus \
+    && ./build.sh \
+    && make install
 
 COPY . /vzlogger
 
-RUN cmake -DBUILD_TEST=off \
- && make \
- && make install
+ARG build_test=off
+RUN cmake -DBUILD_TEST=${build_test} \
+    && make \
+    && make install
 
```

Comment:
Here you also need to add "make test" if test is on, otherwise tests are built but not executed.